Introduce Proxy settings view

jarolrod opened the pull request:
Introduces the components and views that make up the proxy settings page. Missing options handling wiring, as well as the gray box that signals the current status/activation of specified proxy settings. Based on these designs: [proxy designs](https://www.figma.com/file/ek8w3n3upbluw5UL2lGhRx/Bitcoin-Core-App-Design?node-id=6266%3A23933&t=fVYCyV3AXZLRXxpB-4)

| light | dark |
| ----- | ---- |
| <img width="752" alt="Screen Shot 2023-02-17 at 1 15 52 AM" src="https://user-images.githubusercontent.com/23396902/219564274-4156bdff-6d1e-48df-aad1-790800cd1622.png"> | <img width="752" alt="Screen Shot 2023-02-17 at 1 15 28 AM" src="https://user-images.githubusercontent.com/23396902/219564290-2fc83d8c-308d-496a-8de5-34c992701336.png"> |


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/223)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/223)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/223)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/223)

GBKS commented:
Sweet, I'll add this to Figma. Looking at the "IP and port" input, I'd just have kept this as a single input field, I think it's straightforward enough like that. The hit areas are also really small, had to tap a few times until the inputs were selected.

And now (sorry) for some obnoxious feedback from me trying to break things.

Something I noticed is that the port input can accepts line breaks, messing with the layout.

![Screenshot_20230130_104415](https://user-images.githubusercontent.com/695901/215443284-5ada3105-c11e-4d35-83ce-d6d232dcc6b2.jpg)

I can also enter super long text, breaking things further.

![Screenshot_20230130_105435](https://user-images.githubusercontent.com/695901/215445241-7e11ed3d-2b79-4d9a-84cc-45fea9ce2506.jpg)

jarolrod commented:
Updated from[ 8fc193d ](https://github.com/bitcoin-core/gui-qml/commit/8fc193dc52b3085ca8f5bc2528437c07febac71e)to[ af66daa](https://github.com/bitcoin-core/gui-qml/commit/af66daa211b915c018867c030c078d82eb16b6ea), [compare](https://github.com/bitcoin-core/gui-qml/compare/8fc193dc52b3085ca8f5bc2528437c07febac71e..af66daa211b915c018867c030c078d82eb16b6ea)

changes: 
- rebased over main
- addressed conflicts
- partially addressed this [feedback](https://github.com/bitcoin-core/gui-qml/pull/223#issuecomment-1408311647), a follow-up will add a max input length to valueinput

johnny9 reviewed (approved):
ACK af66daa211b915c018867c030c078d82eb16b6ea

Looks good. No qml errors.
